pandas/core/common.py

```diff
@@ -406,7 +406,11 @@ def random_state(state=None):
     -------
     np.random.RandomState
     """
-    if is_integer(state):
+    if (
+        is_integer(state)
+        or is_array_like(state)
+        or isinstance(state, np.random.BitGenerator)
```

Comment:
BitGenerator is somewhat new (NumPy>=1.17 IIRC). You'll need to put this in a conditional check. `pandas.compat` has some helpers.